src/api/register_agent.rs

```diff
@@ -45,13 +44,11 @@ impl Client {
         let mac_addresses: Vec<String> = networks
             .iter()
             .map(|interface| &interface.mac_addr)
-            .filter(|mac_opts| mac_opts.is_some())
-            .map(|mac| mac.clone().unwrap())
+            .filter_map(|mac| mac.clone())
```

Comment:
I don't understand how it's the same that the previous code.
Can you explain it to me ? Thank you

Reply:
because filter_map does both filter and unwrap https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.filter_map

Reply:
Oh ! Thanks for the explanation.

Reply:
And for this part ?
filter(|mac_opts| mac_opts.is_some())

Reply:
filter(|mac_opts| mac_opts.is_some()) -> that filters the null values, filter_map does the same thing + unwrap